csrc/kernel.cpp

```diff
@@ -315,6 +315,11 @@ void Kernel::finalize(std::vector<Expr*> top_level_exprs) {
   summary_.sync_map = GpuLower::current()->syncMap();
   summary_.parallel_dimension_map_ =
       GpuLower::current()->parallelDimensionMap();
+  parameters_ = GpuLower::current()->allKnownVals();
+  parameters_.insert(parameters_.end(), outputs().begin(), outputs().end());
```

Comment:
Should we move the `ValType == TensorView` check for outputs here? There is no check in `Fusion::addOutputs` for scalar outputs. I am assuming it is still unsupported to have scalar outputs, but now that I think of it, the `inputs->kernel_inputs` machinery could also be used for evaluating scalar outputs...